YAML: 1.0

preprocess:
  point_filter_num: 1
  lidar_type: 1 # 1-AVIA 2-velodyne 3-ouster  4-robosense 5-pandar
  blind: 0.1
  sweep_cut_num: 1
  degenerate_window_num: 10

common:
  imu_topic: /livox/imu #/imu_raw   /gps/gtimu
  lid_topic: /livox/lidar #/points_raw  /rslidar_points /velodyne_points
  gnorm: 9.805 # 9.805    # 部分数据，加速度模长为1->9.805gnorm

mapping:
  extrinsic_est_en: true
  extrinsic_T: [0.0, 0.0, -0.0]
  extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]

map_options:
  map_type: MULTI_RESOLUTION_VOXEL_HASHMAP
  resolutions:
    - resolution: 0.2
      max_num_points: 20
      min_distance_between_points: 0.05
    - resolution: 0.5
      max_num_points: 20
      min_distance_between_points: 0.08
    - resolution: 1.2
      max_num_points: 40
      min_distance_between_points: 0.18
  neig_options:
    distance_max: 60
    radius_min: 0.2
    radius_max: 1.2
    exponent: 1.0

delay_time: 0.2

odometry:
  surf_res: 0.1
  log_print: false
  max_num_iteration: 10
  # ct_icp
  icpmodel: CT_POINT_TO_PLANE # Options: [CT_POINT_TO_PLANE, POINT_TO_PLANE]
  size_voxel_map: 0.5 # The voxel size of in the voxel map
  min_distance_points: 0.15
  max_num_points_in_voxel: 20 # The maximum number of points per voxel of the map
  max_distance: 500.0 # The threshold of the distance to suppress voxels from the map
  weight_alpha: 0.9
  weight_neighborhood: 0.1
  max_dist_to_plane_icp: 0.2 # This is important
  init_num_frames: 20
  voxel_neighborhood: 1
  max_number_neighbors: 20
  threshold_voxel_occupancy: 1
  estimate_normal_from_neighborhood: true
  min_number_neighbors: 20 # The minimum number of neighbor points to define a valid neighborhood
  power_planarity: 2.0
  num_closest_neighbors: 1

  sampling_rate: 1.
  max_num_residuals: 1000
  min_num_residuals: 50
  motion_compensation: CONTINUOUS #NONE, CONSTANT_VELOCITY, ITERATIVE, CONTINUOUS
  beta_location_consistency: 0.001
  beta_orientation_consistency: 0.1
  beta_constant_velocity: 0.001
  beta_small_velocity: 0.0

  thres_translation_norm: 0.01 # m
  thres_orientation_norm: 0.1 # deg

  satu_acc: 19.5 # the saturation value of IMU's acceleration. not related to the units
  satu_gyro: 16 # the saturation value of IMU's angular velocity. not related to the units
